File: leetcode/List/leetcode_704_binary_search/Solution.java
package List.leetcode_704_binary_search;
class Solution {
    public int search(int[] nums, int target) {
        int left = 0, right = nums.length;
        while (left < right){
            int mid = (left + right) / 2;
            // int mid = left + (right-left) >> 1; >>means (right - left) /2.
            if (nums[mid] == target)
                return mid;
            else if (nums[mid] < target)
                left = mid + 1;
            else if (nums[mid] > target)
                right = mid;
        }
        return -1;    
    }
}

File: leetcode/List/leetcode_704_binary_search/solution.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        min = 0
        max = len(nums)-1
        while min <= max:
            middle = (min + max) // 2
            if nums[middle] < target:
                min = middle + 1
            elif nums[middle] > target:
                max = middle -1
            elif nums[middle] == target:
                return middle
        return -1

